### PR TITLE
Layout manager

### PR DESCRIPTION
**Pull request by aaronpcz**
Support for managing the layout of the grid and doing some under the hood changes to support best usage of screen space when collapsing and expanding components.

### CODE REVIEW

File: Gulpfile.js [hunk 1]
```diff
@@ -55,10 +55,10 @@ if(debugMode) {
   abstract out the process of making a bundler, and then leave it open to manual triggering
 */
 var indexBundler = bundlerBuilder('./src/js/index.jsx', 'index.js', true);
```
Comment (aaronpcz) [inline Gulpfile.js]: Pretty sure we only need the one bundle now.  If anyone else knows any differently I'll roll back the changes made to the gulpfile.

File: src/js/stores/managers/layout-manager.js [hunk 2]
```diff
@@ -0,0 +1,309 @@
...
+
+var _ = require('lodash');
+var hat = require('hat');
+var ReactGridUtils = require('react-grid-layout').utils;
+var LayoutComponentColumns = require('../../constants/layout-constants').ColumnTypes;
+
+class LayoutManager {
+    constructor() {
+        // APEP TODO set initial state
+        this.layout = [];
```
Comment (aaronpcz) [inline src/js/stores/managers/layout-manager.js]: We may wish to add in some default components like before

File: Gulpfile.js (hunk 1 above)
Comment (Stormgun) [reply Gulpfile.js]: So, at the moment we only need one bundler, as all the components can be accessed in one place.  
I think the way the routing is structured at the moment  does a good job, but I am open to suggestions if we think we might want to create different bundles

File: src/js/stores/managers/layout-manager.js (hunk 2 above)
Comment (Stormgun) [reply src/js/stores/managers/layout-manager.js]: The idea for the default components was to have a base layout that would make sense to a first-time user - to aid this I think that some sort of nice tooltip library can be implemented as a separate feature

Comment (aaronpcz) [reply src/js/stores/managers/layout-manager.js]: Agreed, will add into a commit after we are happy with it.

File: Gulpfile.js (hunk 1 above)
Comment (aaronpcz) [reply Gulpfile.js]: Let's keep with the one for now considering everything is in the single application now anyway

**Pull request merged**